modnet/fit_genetic.py

```diff
@@ -0,0 +1,246 @@
+from random import randint
+import random
+from typing import List, Optional
+import tensorflow.keras as keras
+import pandas as pd
+import numpy as np
+from sklearn.metrics import mean_absolute_error as mae
+from sklearn.metrics import mean_squared_error as mse
+from sklearn.model_selection import train_test_split
+from modnet.preprocessing import MODData
+from modnet.models import MODNetModel
+
+
+class FitGenetic:
+    """Class optimizing the model parameters using a genitic algorithm.
+    """
+
+    def __init__(
+        self,
+        data: MODData,
+        size_pop=15,
+        num_epochs=5,
+        prob_mut=0.8
+        ):
+
+        """Initializes parameters used in this class.
+
+        Parameters:
+            size_pop: Size of the population.
+            num_epochs: Number of generations.
+            prob_mut: Probability the mutation occurs.
+            data: A 'MODData' that has been featurized and feature selected.
+        """
+
+        self.size_pop = size_pop
+        self.num_epochs = num_epochs
+        self.prob_mut = prob_mut
+        self.data = data
+
+
+    def train_val_split(
+        self,
+        data: MODData
+        )->None:
+
+        """Splits arrays or matrices into random train and validation subsets.
+        
+        Parameter:
+            data: 'MODData' data which need to be splitted.
+        """
+
+        split_border = int(0.9*data.df_targets.shape[0])
+        final_border = int(data.df_targets.shape[0])
+        self.X_train, self.X_val = data.split((range(split_border),range(split_border,final_border)))
+        self.y_train = self.X_train.df_targets
+        self.y_val = self.X_val.df_targets
+
+        return self.X_train, self.X_val, self.y_train, self.y_val
+
+
+    def initialization_population(
+        self,
+        size_pop: int
+        )->None:
+
+        """Inintializes the initial population (Generation 0).
+       
+        Paramter:
+            size_pop: Size of the population.
+        """
+
+        self.pop =  [[]]*size_pop
+        activation = ['relu', 'elu']
+        loss = ['mae', 'mse']
+        xscale = ['minmax', 'normal']
+        lr = [0.03, 0.01, 0.003, 0.001, 0.0003, 0.0001]
```

Comment:
Possible suggestion from the meeting: reduce learning rate search space to that of fit preset

```suggestion
        lr = [0.01, 0.005, 0.001]
```